lib/test.js

```diff
@@ -66,12 +70,18 @@ class ExecutionContext {
 		boundPlan.skip = () => {};
 
 		Object.defineProperties(this, assertionNames.reduce((props, name) => {
-			props[name] = {value: assertions[name].bind(test)};
+			const refreshTimeoutOnAssertion = assertion => (...args) => {
+				test.refreshTimeout();
+				return assertion.call(test, ...args);
+			};
+			const assertion = refreshTimeoutOnAssertion(assertions[name]);
```

Comment:
Rather than wrapping the assertions, I think it would be sufficient to add a call to `refreshTimeout()` in `countPassedAssertion()`, `addPendingAssertion()` and `addFailedAssertion()`. The timer should be cleared once again when the pending assertion resolves.